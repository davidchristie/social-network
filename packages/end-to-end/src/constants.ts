export const ORIGIN = "http://localhost:3000";
export const USER_1_EMAIL = "test_user1@email.com";
export const USER_1_PASSWORD = "test_user1password";
