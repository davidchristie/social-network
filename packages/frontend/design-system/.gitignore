.cache/
.rpt2_cache/
documentation/
