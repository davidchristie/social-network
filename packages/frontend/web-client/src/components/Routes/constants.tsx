import React from "react";

export interface RouteData {
  load: () => Promise<{
    default: React.ComponentType
  }>;
  onlyAuthenticated?: boolean;
  onlyUnauthenticated?: boolean;
  path: string;
}

export const defaultAuthenticatedPath = "/";
export const defaultUnauthenticatedPath = "/login";

export const routes: RouteData[] = [
  {
    load: () => import(/* webpackChunkName: "HomePage" */ "../HomePage"),
    onlyAuthenticated: true,
    path: "/",
  },
  {
    load: () => import(/* webpackChunkName: "AccountPage" */ "../AccountPage"),
    onlyAuthenticated: true,
    path: "/account",
  },
  {
    load: () => import(/* webpackChunkName: "LoginPage" */ "../LoginPage"),
    onlyUnauthenticated: true,
    path: "/login",
  },
  {
    load: () => import(/* webpackChunkName: "ProfilePage" */ "../ProfilePage"),
    path: "/profile/:id",
  },
  {
    load: () => import(/* webpackChunkName: "SignupPage" */ "../SignupPage"),
    onlyUnauthenticated: true,
    path: "/signup",
  },
];
