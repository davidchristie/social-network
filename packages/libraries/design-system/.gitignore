.rpt2_cache
documentation/
