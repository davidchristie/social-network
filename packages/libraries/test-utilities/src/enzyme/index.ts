export { default as beforeEachWaitForUpdate } from "./beforeEachWaitForUpdate";
export { default as itContainsComponent } from "./itContainsComponent";
