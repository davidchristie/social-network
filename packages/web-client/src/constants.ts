export const AUTHENTICATION_TOKEN = "authentication_token";
