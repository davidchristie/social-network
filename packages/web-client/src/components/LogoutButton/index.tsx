import React from "react";

import Button from "../Button";
import { AUTHENTICATION_TOKEN } from '../../constants'

const handleClick = () => {
  window.localStorage.removeItem(AUTHENTICATION_TOKEN);
  window.location.reload();
};

export default class LogoutButton extends React.Component {
  public render () {
    return (
      <Button onClick={handleClick}>
        Logout
      </Button>
    );
  }
}
