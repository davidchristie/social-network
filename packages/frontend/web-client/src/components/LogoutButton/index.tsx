import { Button } from "design-system";
import React from "react";

import { AUTHENTICATION_TOKEN } from "../../constants";

const handleClick = () => {
  window.localStorage.removeItem(AUTHENTICATION_TOKEN);
  window.location.reload();
};

export default class LogoutButton extends React.Component {
  public render () {
    return (
      <Button onClick={handleClick}>
        Logout
      </Button>
    );
  }
}
