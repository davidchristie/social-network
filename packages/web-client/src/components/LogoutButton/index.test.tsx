import { Button } from "design-system";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

import { AUTHENTICATION_TOKEN } from "../../constants";
import LogoutButton from "./index";

describe("LogoutButton component", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<LogoutButton />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("when clicked", () => {
    beforeEach(() => {
      window.localStorage.setItem(AUTHENTICATION_TOKEN, "test_token");
      Object.defineProperty(window.location, "reload", {
        configurable: true,
      });
      window.location.reload = jest.fn();
      wrapper.find(Button).simulate("click");
    });

    it("removes authentication token from local storage", () => {
      expect(window.localStorage.getItem(AUTHENTICATION_TOKEN)).toBeNull();
    });

    it("reloads the page", () => {
      expect(window.location.reload).toHaveBeenCalledTimes(1);
    });
  });
});
