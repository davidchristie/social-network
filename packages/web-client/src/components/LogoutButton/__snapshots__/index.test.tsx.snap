// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutButton component matches snapshot 1`] = `
<Button
  onClick={[Function]}
>
  Logout
</Button>
`;
